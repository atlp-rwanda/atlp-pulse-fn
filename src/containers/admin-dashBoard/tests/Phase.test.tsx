import { MockedProvider as ApolloProvider } from '@apollo/client/testing';
import '@testing-library/jest-dom';
import React from 'react';
import '../../../../test/jest/__mocks__/matchMedia';

import { act, fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AdminPhases from '../Phases';

describe('Phases tests', () => {
  it('Snapshots Phases Page', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <ApolloProvider>
            <AdminPhases />
          </ApolloProvider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  jest.setTimeout(5000);
  it('Renders Phases Page', async () => {
    render(
      <MemoryRouter>
        <ApolloProvider addTypename={false}>
          <AdminPhases />
        </ApolloProvider>
      </MemoryRouter>,
    );

    const updateIcon = await screen.findByTestId('updateIcon');
    const deleteIcon = await screen.findByTestId('deleteIcon');
    expect(updateIcon).toBeInTheDocument();
    expect(deleteIcon).toBeInTheDocument();

    await act(async () => {
      fireEvent.click(updateIcon);
    });
    const confirmUpdateBtn = await screen.findByTestId('confirmUpdateBtn');
    expect(confirmUpdateBtn).toBeInTheDocument();
    await act(async () => {
      fireEvent.click(confirmUpdateBtn);
    });

    await act(async () => {
      fireEvent.click(deleteIcon);
    });
    const confirmDeleteBtn = await screen.findByTestId('confirmDeleteBtn');
    expect(confirmDeleteBtn).toBeInTheDocument();
    await act(async () => {
      fireEvent.click(confirmDeleteBtn);
    });
  });
});
