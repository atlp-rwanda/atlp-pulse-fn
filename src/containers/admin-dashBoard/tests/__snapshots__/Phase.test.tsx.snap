// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phases tests Snapshots Phases Page 1`] = `
Array [
  <div
    className="h-screen w-screen bg-black bg-opacity-30 backdrop-blur-sm fixed top-0 left-0 z-20 flex items-center justify-center px-4 hidden"
  >
    <div
      className="bg-indigo-100 dark:bg-dark-bg w-screen md:w-1/2  xl:w-4/12 rounded-lg p-4 pb-8"
    >
      <div
        className="card-title w-full flex  flex-wrap justify-center items-center  "
      >
        <h3
          className="font-bold text-sm dark:text-white text-center w-11/12 uppercase"
        >
          Add Phase
        </h3>
        <hr
          className=" bg-primary border-gray-400 my-3 w-full"
        />
      </div>
      <div
        className="card-body"
      >
        <form
          className=" py-3 px-8"
        >
          <div
            className="input my-5 h-9 "
          >
            <div
              className="grouped-input flex items-center h-full w-full rounded-md"
            >
              <input
                className="border border-primary rounded outline-none px-5 dark:bg-dark-bg-frame dark:text-white font-sans text-xs py-2 w-full"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="name"
                type="text"
              />
            </div>
          </div>
          <div
            className="input my-5 h-9 "
          >
            <div
              className="grouped-input flex items-center h-full w-full rounded-md"
            >
              <input
                className="border border-primary py-2 dark:bg-dark-bg-frame dark:text-white rounded outline-none px-5 font-sans text-xs w-full"
                name="description"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Description"
                type="text"
              />
            </div>
          </div>
          <div
            className="w-full flex justify-between"
          >
            <button
              className="btn info sm w-[30%] md:w-1/4 text-sm font-sans"
              data-testid="remove"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
               
              cancel
            </button>
            <button
              className="btn primary sm w-[30%] md:w-1/4 text-sm font-sans p-0"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
               
              save
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  <div
    className="h-screen w-screen bg-black fixed top-0 left-0 z-20 bg-opacity-30 backdrop-blur-sm flex items-center justify-center overflow-auto p-4 hidden"
  >
    <div
      className="w-screen p-4 pb-8 bg-indigo-100 rounded-lg dark:bg-dark-bg md:w-1/2 xl:w-4/12"
    >
      <div
        className="flex flex-wrap items-center justify-center w-full card-title "
      >
        <h3
          className="w-11/12 text-sm font-bold text-center uppercase dark:text-white"
        >
          Update Phase
        </h3>
        <hr
          className="w-full my-3 border-gray-400  bg-primary"
        />
      </div>
      <div
        className="card-body"
      >
        <form
          className="px-8 py-3 "
        >
          <div
            className="my-5 input h-9 "
          >
            <div
              className="flex items-center w-full h-full rounded-md grouped-input"
            >
              <input
                className="w-full px-5 py-2 font-sans text-xs border rounded outline-none border-primary dark:bg-dark-bg-frame dark:text-white"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="name"
                type="text"
              />
            </div>
          </div>
          <div
            className="my-5 input h-9 "
          >
            <div
              className="flex items-center w-full h-full rounded-md grouped-input"
            >
              <input
                className="w-full px-5 py-2 font-sans text-xs border rounded outline-none border-primary dark:bg-dark-bg-frame dark:text-white"
                name="description"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Description"
                type="text"
              />
            </div>
          </div>
          <div
            className="flex justify-between w-full"
          >
            <button
              className="btn info sm w-[30%] md:w-1/4 text-sm font-sans"
              data-testid="remove"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
               
              cancel
            </button>
            <button
              className="btn primary sm w-[30%] md:w-1/4 text-sm font-sans p-0"
              data-testid="confirmUpdateBtn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
               
              save
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  <div
    className="h-screen w-screen bg-black fixed top-0 left-0 z-20 bg-opacity-30 backdrop-blur-sm flex items-center justify-center overflow-auto p-4 hidden"
  >
    <div
      className="bg-indigo-100 dark:bg-dark-bg w-screen md:w-1/2  xl:w-4/12 rounded-lg p-4 pb-8"
    >
      <div
        className="card-title w-full flex  flex-wrap justify-center items-center  "
      >
        <h3
          className="font-bold text-xl dark:text-white text-center w-11/12"
        >
          Delete Phase
        </h3>
        <hr
          className=" bg-primary border-gray-400 my-3 w-full"
        />
      </div>
      <div
        className="card-body"
      >
        <form
          className=" py-3 px-8"
        >
          <div>
            <h2
              className="text-base dark:text-white text-center m-4"
            >
              do you really want to remove this Phase?
            </h2>
          </div>
          <div
            className="w-full flex justify-between"
          >
            <button
              className="btn info sm w-[30%] md:w-1/4 text-sm font-sans"
              data-testid="delete"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
            <button
              className="btn danger sm w-[30%] md:w-1/4 text-sm font-sans"
              data-testid="confirmDeleteBtn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Delete
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  <div
    className="bg-light-bg dark:bg-dark-frame-bg"
  >
    <div
      className="flex items-right pb-8"
    >
      <div
        className="flex gap-2"
      >
        <button
          className="btn primary lg m-0"
          data-testid="removeModel"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
           
          Phase
           
            
          +
        </button>
      </div>
    </div>
    <div
      className=""
    >
      <div
        className="bg-indigo-100 dark:bg-dark-bg shadow-lg px-5 py-8 rounded-md w-[100%] \\"lg:ml-60 mx-auto\\"} mb-10"
      >
        <div
          className="flex items-center justify-between pb-6 "
        >
          <div>
            <h2
              className="text-xl font-semibold text-gray-800 dark:text-white "
            >
              Phase list
            </h2>
            <input
              className="px-5 py-2 mt-4 font-sans text-xs border border-primary rounded outline-none dark:bg-neutral-600 dark:text-white w-52 md:w-96"
              defaultValue=""
              onChange={[Function]}
              placeholder="Filter"
            />
          </div>
        </div>
        <div
          style={
            Object {
              "overflowX": "auto",
            }
          }
        >
          <table
            className="min-w-full leading-normal"
            role="table"
          >
            <thead>
              <tr
                role="row"
              >
                <th
                  className=" thead"
                  colSpan={1}
                  onClick={[Function]}
                  role="columnheader"
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                  title="Toggle SortBy"
                >
                  name
                </th>
                <th
                  className=" thead"
                  colSpan={1}
                  onClick={[Function]}
                  role="columnheader"
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                  title="Toggle SortBy"
                >
                  Description
                </th>
                <th
                  className=" thead"
                  colSpan={1}
                  onClick={[Function]}
                  role="columnheader"
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                  title="Toggle SortBy"
                >
                  Actions
                </th>
              </tr>
            </thead>
            <tbody
              role="rowgroup"
            >
              <tr
                className=" bg-white dark:bg-dark-bg "
                role="row"
              >
                <td
                  className="data-cell"
                  role="cell"
                />
                <td
                  className="data-cell"
                  role="cell"
                />
                <td
                  className="data-cell"
                  role="cell"
                >
                  <div
                    className="flex relative flex-row align-middle items-center"
                  >
                    <div
                      data-testid="updateIcon"
                      onClick={[Function]}
                    >
                      <span />
                    </div>
                    <div
                      data-testid="deleteIcon"
                      onClick={[Function]}
                    >
                      <span />
                    </div>
                  </div>
                </td>
              </tr>
              <tr>
                <td
                  colSpan={3}
                  style={
                    Object {
                      "textAlign": "center",
                    }
                  }
                >
                  loading...
                </td>
              </tr>
            </tbody>
          </table>
          <div>
            <table
              className="w-full mt-2"
            >
              <tfoot
                className="w-full py-2"
              >
                <tr
                  className="w-full py-2"
                >
                  <td
                    colSpan={3}
                  >
                    <div
                      className="w-full justify-center flex mx-auto flex-row items-center overflow-x-auto"
                    >
                      <button
                        className="page mx-2 text-white rounded-circle appearance-none font-bold flex items-center justify-center bg-primary h-[30px] w-[60px] cursor-pointer"
                        disabled={true}
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          className="w-5"
                          fill="none"
                          fontSize="sm"
                          stroke="currentColor"
                          strokeWidth={2}
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M11 15l-3-3m0 0l3-3m-3 3h8M3 12a9 9 0 1118 0 9 9 0 01-18 0z"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                          />
                        </svg>
                      </button>
                       
                      <button
                        className="page mx-2 text-white rounded-circle font-bold flex items-center justify-center bg-primary h-[30px] w-[60px] cursor-pointer"
                        disabled={true}
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          className="w-5"
                          fill="none"
                          fontSize="sm"
                          stroke="currentColor"
                          strokeWidth={2}
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                          />
                        </svg>
                      </button>
                       
                      <div
                        className="flex flex-row justify-center w-full sm:text-[12px] items-center "
                      >
                        <span
                          className="inline-block mx-2"
                        >
                          Page
                           
                          <strong>
                            1
                             of
                             1
                          </strong>
                           
                        </span>
                        <span
                          className="inline-block mx-2"
                        >
                          | Go to page:
                           
                          <input
                            className=" outline-none border rounded-md appearance-none border-primary dark:bg-primary pl-1"
                            defaultValue={1}
                            onChange={[Function]}
                            style={
                              Object {
                                "border": "solid 0.1rem #9e85f5",
                                "height": "30px",
                                "paddingLeft": "1.2rem",
                                "width": "50px",
                              }
                            }
                            type="number"
                          />
                        </span>
                         
                        <select
                          className="px-1/2 font-raleway rounded-md border border-primary dark:bg-primary"
                          onChange={[Function]}
                          style={
                            Object {
                              "border": "solid 0.1rem #9e85f5",
                              "height": "30px",
                            }
                          }
                          value={3}
                        >
                          <option
                            value={3}
                          >
                            Show 
                            3
                          </option>
                          <option
                            value={5}
                          >
                            Show 
                            5
                          </option>
                          <option
                            value={10}
                          >
                            Show 
                            10
                          </option>
                          <option
                            value={25}
                          >
                            Show 
                            25
                          </option>
                          <option
                            value={50}
                          >
                            Show 
                            50
                          </option>
                          <option
                            value={100}
                          >
                            Show 
                            100
                          </option>
                        </select>
                      </div>
                    </div>
                  </td>
                </tr>
              </tfoot>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
